src/ray/raylet/node_manager.cc

```diff
@@ -1172,6 +1182,49 @@ void NodeManager::DisconnectClient(const std::shared_ptr<ClientConnection> &clie
   // these can be leaked.
 }
 
+void NodeManager::DeleteLocalURI(const std::string &uri, std::function<void(bool)> cb) {
+  auto random_id = UniqueID::FromRandom();
+  auto resource_path = boost::filesystem::path(initial_config_.resource_dir);
+  std::string sep = "://";
+  auto pos = uri.find(sep);
+  if (pos == std::string::npos || pos + sep.size() == uri.size()) {
+    RAY_LOG(ERROR) << "Invalid uri: " << uri;
+    cb(true);
+  }
+
+  auto from_path = resource_path / boost::filesystem::path(uri.substr(pos + sep.size()));
+  if (!boost::filesystem::exists(from_path)) {
+    RAY_LOG(ERROR) << uri << " doesn't exist locally: " << from_path;
+    cb(true);
+  }
+  auto to_path = resource_path / boost::filesystem::path(random_id.Hex());
```

Comment:
Instead of a random path, how about something deterministic like <path>.to_delete?

Reply:
I make it a random one is to avoid a race condition. Maybe I can check file existence and add a suffix for this.